tests/ProxyManagerTest/GeneratorStrategy/FileWriterGeneratorStrategyTest.php

```diff
@@ -130,4 +133,117 @@ public function testWhenFailingAllTemporaryFilesAreRemoved() : void
+
+        $configuration
+            ->expects(self::any())
+            ->method('getProxiesTargetDir')
+            ->willReturn($proxyFolder);
+
+        $proxyAutoloader
+            ->expects(self::once())
+            ->method('__invoke')
+            ->with($proxyClassName)
+            ->will(self::returnCallback(function ($className) : bool {
+                // We does not include file here.
+                return true;
+            }));
+
+        spl_autoload_register(function ($name) use ($fqcn) {
+            if ($name === $fqcn) {
+                throw new \Exception(sprintf("Cannot load class %s", $name));
+            }
+        });
+
+        $locator
+            ->expects(self::any())
+            ->method('getProxyFileName')
+            ->with($fqcn)
+            ->will(self::returnValue($tmpFile));
+
+        $body = $generator->generate(new ClassGenerator($fqcn));
+
+
+        self::assertFileExists($tmpFile);
+        self::expectException(\Exception::class);
+        self::expectExceptionMessage(sprintf("Cannot load class %s", $fqcn));
+        $generator->classExists($proxyClassName, $configuration);
+        rmdir($tmpFile);
+    }
+
+    public function testClassNotExistsLoaded(): void
+    {
+        /* @var $locator FileLocatorInterface|\PHPUnit_Framework_MockObject_MockObject */
+        $locator   = $this->createMock(FileLocatorInterface::class);
+        $generator = new FileWriterGeneratorStrategy($locator);
+        $namespace = 'Foo';
+        $className = UniqueIdentifierGenerator::getIdentifier('Bar');
+        $fqcn      = $namespace . '\\' . $className;
+        $proxyClassName = $fqcn;
+        $proxyFolder = sys_get_temp_dir() . '/';
+        $tmpFile   = $proxyFolder . $namespace . $className . '.php';
+        $configuration = $this->createMock(Configuration::class);
+        $proxyAutoloader = $this->createMock(AutoloaderInterface::class);
+
+        $configuration
+            ->expects(self::any())
+            ->method('getProxyAutoloader')
+            ->will(self::returnValue($proxyAutoloader));
+
+        $configuration
+            ->expects(self::any())
+            ->method('getProxiesTargetDir')
+            ->willReturn($proxyFolder);
+
+        $proxyAutoloader
+            ->expects(self::once())
+            ->method('__invoke')
+            ->with($proxyClassName)
+            ->will(self::returnCallback(function ($className) use ($fqcn, $tmpFile) : bool {
+                if ($fqcn === $className) {
+                    /* @noinspection PhpIncludeInspection */
+                    /* @noinspection UsingInclusionOnceReturnValueInspection */
+                    require_once $tmpFile;
+                }
+
+                return true;
+            }));
+
+        spl_autoload_register(function ($name) use ($fqcn) {
+            if ($name === $fqcn) {
+                throw new \Exception(sprintf("Cannot load class %s", $name));
```

Comment:
Oh, now I understand what the bug is about. I don't think this needs fixing, tbh - this is a broken autoloader, not a broken ProxyManager detail.

The simplest solution for this is to have the proxy autoloader on top of the autoload stack, but overall the sanest solution is to trash the broken autoloader. Autoloaders should have a `:bool` signature and only crash on fatals...

Reply:
@Ocramius. Thanks for comment. I understand this, but many old projects has "bad" autoloader, and this fix can make them work. In another side if proxies does not generated before  script run we will have error even if the proxy autoloader are registered at top. ZF1 one of "wrong" loader examples. Unfortunately not all projects has composer autoloader or has custom autoloader that followed by best practices.

Reply:
With this patch they will generate proxy if they don't generated for "file write" strategy without the work break.